libraries/AP_Scripting/examples/Simple_Serial_Dump.lua

```diff
@@ -0,0 +1,35 @@
+-- this script reads data from a serial port and dumps it to a file
+
+local file_name = 'raw serial dump.txt'
+local baud_rate = 230400
+
+-- find the serial first (0) scripting serial port instance
+-- SERIALx_PROTOCOL 28
+local port = assert(serial:find_serial(0),"Could not find Scripting Serial Port")
+
+-- make a file
+local file = assert(io.open(file_name, "w"),"Could not create file " .. file_name)
+file:close()
+
+-- begin the serial port
+port:begin(baud_rate)
+port:set_flow_control(0)
+
+function update() -- this is the loop which periodically runs
+
+  local n_bytes = math.min(port:available(), 512)
+  file = io.open(file_name, "a")
+  while n_bytes > 0 do
+      c = port:read()
+      if c == nil then
+          break
+      end
+      file:write(string.char())
```

Comment:
```suggestion
      file:write(string.char(c))
```

Reply:
Question.

How was this working with this problem in here?!

The data coming out looked valid to me (it's in my LD06 PR).

Reply:
I don't know, that would imply a really interesting bug in the Lua interpreter. I couldn't find the PR you mentioned, but I assume you fixed it and forgot to update the code in Git.

Also `read` returns -1 on error, not `nil`.

And it's strange to open and close the file every loop. Probably better to simply flush it every n loops.